From 8184c44fda736e589e38b044e487cb16a969eb9e Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Mon, 18 Nov 2024 15:05:22 +0100
Subject: [PATCH 1376/1695] dts: stm32: rename scmi regulators by regulator@0
 for stm32mp25x boards

Use a scmi regulator node name coherent with scmi binding:
protocol@17{
	regulators {
		regulator@0 {
		};
		[...]
	};
};

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f87ae5a9a01b8b61482fbbfc260cbda237a72ff9 dts: stm32: rename scmi regulators by regulator@0 for stm32mp25x boards]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: Iff76581f19e0768e0cbb74ce7b1760f131a38265
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/419309
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp257f-dk.dts  | 73 +++++++++++++++------------
 core/arch/arm/dts/stm32mp257f-ev1.dts | 55 +++++++++++---------
 2 files changed, 72 insertions(+), 56 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index 9397e733c..7ccdda94b 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -413,38 +413,47 @@
 	status = "okay";
 };
 
-&scmi_regu {
-	voltd-vddcore {
-		reg = <VOLTD_SCMI_STPMIC2_BUCK2>;
-		voltd-supply = <&vddcore>;
-	};
-	voltd-v1v8 {
-		reg = <VOLTD_SCMI_STPMIC2_BUCK5>;
-		voltd-supply = <&v1v8>;
-	};
-	voltd-v3v3 {
-		reg = <VOLTD_SCMI_STPMIC2_BUCK7>;
-		voltd-supply = <&v3v3>;
-	};
-	voltd-v5v-hdmi {
-		reg = <VOLTD_SCMI_STPMIC2_LDO5>;
-		voltd-supply = <&v5v_hdmi>;
-	};
-	voltd-v5v-vconn {
-		reg = <VOLTD_SCMI_STPMIC2_LDO6>;
-		voltd-supply = <&v5v_vconn>;
-	};
-	voltd-vdd-emmc {
-		reg = <VOLTD_SCMI_STPMIC2_LDO2>;
-		voltd-supply = <&vdd_emmc>;
-	};
-	voltd-vdd3v3-usb {
-		reg = <VOLTD_SCMI_STPMIC2_LDO4>;
-		voltd-supply = <&vdd3v3_usb>;
-	};
-	voltd-vdd-sdcard {
-		reg = <VOLTD_SCMI_STPMIC2_LDO7>;
-		voltd-supply = <&vdd_sdcard>;
+&scmi_regulator {
+	regulators {
+		regulator@b {
+			reg = <VOLTD_SCMI_STPMIC2_BUCK2>;
+			voltd-supply = <&vddcore>;
+		};
+
+		regulator@e {
+			reg = <VOLTD_SCMI_STPMIC2_BUCK5>;
+			voltd-supply = <&v1v8>;
+		};
+
+		regulator@10 {
+			reg = <VOLTD_SCMI_STPMIC2_BUCK7>;
+			voltd-supply = <&v3v3>;
+		};
+
+		regulator@15 {
+			reg = <VOLTD_SCMI_STPMIC2_LDO5>;
+			voltd-supply = <&v5v_hdmi>;
+		};
+
+		regulator@16 {
+			reg = <VOLTD_SCMI_STPMIC2_LDO6>;
+			voltd-supply = <&v5v_vconn>;
+		};
+
+		regulator@12 {
+			reg = <VOLTD_SCMI_STPMIC2_LDO2>;
+			voltd-supply = <&vdd_emmc>;
+		};
+
+		regulator@14 {
+			reg = <VOLTD_SCMI_STPMIC2_LDO4>;
+			voltd-supply = <&vdd3v3_usb>;
+		};
+
+		regulator@17 {
+			reg = <VOLTD_SCMI_STPMIC2_LDO7>;
+			voltd-supply = <&vdd_sdcard>;
+		};
 	};
 };
 
diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index 45ea74d5b..cd997bf55 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -441,30 +441,37 @@
 	status = "okay";
 };
 
-&scmi_regu {
-	voltd-vddcore {
-		reg = <VOLTD_SCMI_STPMIC2_BUCK2>;
-		voltd-supply = <&vddcore>;
-	};
-	voltd-v1v8 {
-		reg = <VOLTD_SCMI_STPMIC2_BUCK5>;
-		voltd-supply = <&v1v8>;
-	};
-	voltd-v3v3 {
-		reg = <VOLTD_SCMI_STPMIC2_BUCK7>;
-		voltd-supply = <&v3v3>;
-	};
-	voltd-vdd-emmc {
-		reg = <VOLTD_SCMI_STPMIC2_LDO2>;
-		voltd-supply = <&vdd_emmc>;
-	};
-	voltd-vdd3v3-usb {
-		reg = <VOLTD_SCMI_STPMIC2_LDO4>;
-		voltd-supply = <&vdd3v3_usb>;
-	};
-	voltd-vdd-sdcard {
-		reg = <VOLTD_SCMI_STPMIC2_LDO7>;
-		voltd-supply = <&vdd_sdcard>;
+&scmi_regulator {
+	regulators {
+		regulator@b {
+			reg = <VOLTD_SCMI_STPMIC2_BUCK2>;
+			voltd-supply = <&vddcore>;
+		};
+
+		regulator@e {
+			reg = <VOLTD_SCMI_STPMIC2_BUCK5>;
+			voltd-supply = <&v1v8>;
+		};
+
+		regulator@10 {
+			reg = <VOLTD_SCMI_STPMIC2_BUCK7>;
+			voltd-supply = <&v3v3>;
+		};
+
+		regulator@12 {
+			reg = <VOLTD_SCMI_STPMIC2_LDO2>;
+			voltd-supply = <&vdd_emmc>;
+		};
+
+		regulator@14 {
+			reg = <VOLTD_SCMI_STPMIC2_LDO4>;
+			voltd-supply = <&vdd3v3_usb>;
+		};
+
+		regulator@17 {
+			reg = <VOLTD_SCMI_STPMIC2_LDO7>;
+			voltd-supply = <&vdd_sdcard>;
+		};
 	};
 };
 
-- 
2.47.2

